tests/test_main.py

```diff
@@ -26,84 +27,139 @@ def test_knowledge_panel_badendpoint():
-                            "text_element": [],
-                            "source_url": "https://world.openfoodfacts.org/country/India/origin/None/data-quality.json",
-                            "description": "data-quality issues related to origin based for India",
+                            "total_issues": 173,
+                            "text_element": '<ul><li><a href="https://au.openfoodfacts.org/data-quality/ecoscore-production-system-no-label">25222 products with ecoscore-production-system-no-label</a></li>\n<li><a href="https://au.openfoodfacts.org/data-quality/ecoscore-origins-of-ingredients-origins-are-100-percent-unknown">23731 products with ecoscore-origins-of-ingredients-origins-are-100-percent-unknown</a></li>\n<li><a href="https://au.openfoodfacts.org/data-quality/no-packaging-data">22649 products with no-packaging-data</a></li></ul>',
                         }
-                    ]
+                    ],
+                    "source_url": "https://au-en.openfoodfacts.org/data-quality",
+                    "description": "This is a data-quality issues related to Australia",
                 }
-            }
+            },
         ]
     }
 
 
-def test_knowledge_panel_with_allergen():
-    assert knowledge_panel(facet_name="allergen") == {
+def test_knowledge_panel_facet_value_and_country(monkeypatch):
+    class MockResponse(object):
+        def __init__(self):
+            self.url = "https://at-en.openfoodfacts.org/allergen/nuts/data-quality.json"
+
+        def json(self):
+            return {
+                "count": 96,
+                "tags": [
+                    {
+                        "id": "en:ecoscore-origins-of-ingredients-origins-are-100-percent-unknown",
+                        "known": 0,
+                        "name": "ecoscore-origins-of-ingredients-origins-are-100-percent-unknown",
+                        "products": 316,
+                        "url": "https://at-en.openfoodfacts.org/allergen/nuts/data-quality/ecoscore-origins-of-ingredients-origins-are-100-percent-unknown",
+                    },
+                    {
+                        "id": "en:ingredients-percent-analysis-ok",
+                        "known": 0,
+                        "name": "ingredients-percent-analysis-ok",
+                        "products": 276,
+                        "url": "https://at-en.openfoodfacts.org/allergen/nuts/data-quality/ingredients-percent-analysis-ok",
+                    },
+                    {
+                        "id": "en:ecoscore-production-system-no-label",
+                        "known": 0,
+                        "name": "ecoscore-production-system-no-label",
+                        "products": 274,
+                        "url": "https://at-en.openfoodfacts.org/allergen/nuts/data-quality/ecoscore-production-system-no-label",
+                    },
+                ],
+            }
+
+    def mock_get(quality_url):
+        return MockResponse()
```

Comment:
you initialized a url, which is a good idea but you don't check it's the one thats used.

In all your mock (not only this one), you should add an assert:

```suggestion
        response = MockResponse()
        assert quality_url == response.url
        return response
```